src/main/resources/lib/hudson/test/js/failureSummary.js

```diff
@@ -24,24 +27,30 @@ function hideFailureSummary(summaryId) {
     document.getElementById(summaryId + HIDELINK_SUFFIX).style.display = "none";
 }
 
+function initializeShowHideLinks(container) {
+    container = container || document;
 
-document.addEventListener('DOMContentLoaded', () => {
-    const testShowlinks = document.querySelectorAll("a[id*=test-][id*=-showlink]");
-    testShowlinks.forEach((element) => {
-        element.addEventListener('click', (event) => {
-            const id = element.id.replace(PREFIX, '').replace(SHOWLINK_SUFFIX, '');
-            const summaryId = PREFIX + id;
-            showFailureSummary(summaryId, document.URL + id + "summary");
-        })
+    container.querySelectorAll('a[id$="-showlink"], a[id$="-hidelink"]').forEach(link => {
+        if (!link.onclick) {
+            link.onclick = handleShowHideClick;
```

Comment:
please use `addEventListener` like the code removed was using, see https://stackoverflow.com/questions/6348494/addeventlistener-vs-onclick

Reply:
Sure - I just made this correction in the latest commit.

Reply:
(Disclaimer: Still haven't run this.) This change request looks wrong since unlike the earlier code, the events are re-initialized on HTTP response without removing previous event listeners? What happens after a few `showFailureSummary`, does that collect event listeners?

Reply:
Hi @daniel-beck 

If I understand correctly, this shouldn't be an issue with the new change:

```
container.querySelectorAll('a[id$="-showlink"], a[id$="-hidelink"]').forEach(link => {
        link.addEventListener('click', handleShowHideClick);
        link.style.cursor = 'pointer';
```

since duplicate event listeners cannot be added to elements [see docs](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#:~:text=If%20the%20function%20or%20object%20is%20already%20in%20the%20list%20of%20event%20listeners%20for%20this%20target%2C%20the%20function%20or%20object%20is%20not%20added%20a%20second%20time.)

https://stackoverflow.com/questions/10364298/what-does-calling-addeventlister-twice-do

Is this what you were referring to?

Thanks

Reply:
Ah, good point, I had only looked at the previous reference. So this won't cause trouble then. Thanks for the link!